atomos.framework/src/main/java/org/atomos/framework/modules/ModuleConnectLoader.java

```diff
@@ -67,36 +140,82 @@ void initEdges(Module module, Configuration loaderConfig, List<ModuleLayer> pare
      */
     @Override
     protected URL findResource(String moduleName, String name) throws IOException {
-    	// TODO
-    	// The moduleName is the module which is trying to find the resource.
-    	// If the module name is not equal to this loaders module name, return null
-    	
-    	// Otherwise use the reader to find the named resource; should protect with doPriv
-    	return null;
+        URL resource=null;
+        if (!this.resolvedModule.name().equals(moduleName)) {
+            resource=null;
+        }
+        else {
+            try {
+                resource = AccessController.doPrivileged((PrivilegedExceptionAction<URL>)(() -> {
+                    ModuleReader m = this.resolvedModule.reference().open();
+                    URI rURI = this.reader.find(name).orElse(null);
+                    return rURI==null ? null : rURI.toURL();
+                }));
+            } catch (PrivilegedActionException e) {
+                if (e.getCause() instanceof IOException) {
+                    throw (IOException) e.getCause();
+                }
+                else {
+                    throw new IOException(e);
+                }
+            }
+        }
+    	return resource;
     }
 
+    
     @Override
     public URL findResource(String name) {
-        // get the package name of the resource
-    	// check if the ModuleDescriptor for the module reference contains the package name in its packages()
-    	// If not return null
-    	// otherwise use findResource(String, String) with the assumption that this module is driving the find
-    	// If the resource does not end in ".class" then check if the package is open for reflection.
-    	// non-class resources should be returned by this method if the package is open
-    	return null;
+    	URL retVal = null;
```

Comment:
My original comments mislead you in the implementation here:

If the package name is contained in the java.lang.module.ModuleDescriptor.packages() set then
- check the findResource(String, String) method using the module name associated with the loader
- if a resource is found then do a check for ".class" extension or if the package is open and if so then
- return the resource

If the package name is NOT contained in the java.lang.module.ModuleDescriptor.packages() set then
- simply return the result of findResource(String, String) method using the module name associated with the loader